### ISSUE TITLE
How to add a package with a specific commit hash?

### ISSUE THREAD

**Issue by Vandaahl** [jetpack-io/devbox#412]
I read the [documentation](https://www.jetpack.io/devbox/docs/guides/pinning_packages/) on this subject but I don't understand how it works. I would like to add [version 16.17.0 of nodejs](https://lazamar.co.uk/nix-versions/?package=nodejs&version=16.17.0&fullName=nodejs-16.17.0&keyName=nodejs-16_x&revision=ee01de29d2f58d56b1be4ae24c24bd91c5380cea&channel=nixpkgs-unstable#instructions) from the nixpkgs-unstable channel. I should add hash ee01de29d2f58d56b1be4ae24c24bd91c5380cea to my devbox.json file, but where? Under packages? I tried it like this but it does not work:

```
{
  "packages": [
    "mariadb",
    "git",
    "php81",
    "php81Packages.composer",
    "apacheHttpd",
    "ee01de29d2f58d56b1be4ae24c24bd91c5380cea"
  ],
  "nixpkgs": {
    "commit": "1fc0a0f6c5290ce80b380d1eea3681a04aea1184"
  }
}

**Comment by Lagoja** [comment]
Hey @Vandaahl, thanks for trying Jetpack! The commit hash should go in the `nixpkgs` object. Like so:

```
{
  "packages": [
    "mariadb",
    "git",
    "php81",
    "php81Packages.composer",
    "apacheHttpd",
  ],
  "nixpkgs": {
    "commit": "ee01de29d2f58d56b1be4ae24c24bd91c5380cea"
  }
}
```

This will configure Devbox to use that version of the Nix store, which will include version 16.17 of NodeJS. I can update the docs to make this clearer. Please let us know if this fixes the issue!

**Comment by Vandaahl** [comment]
Thank you @Lagoja, I think I understand a little better now. The commit hash influences all packages I have listed under "packages". I modified my devbox.json file as you suggested and added nodejs-16_x to packages. This results in the installation of nodejs 16.17.0, which is the exact version I wanted. As a side effect it has also downgraded the other packages I have listed, but that's ok. Problem solved!
